sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -617,13 +617,18 @@ class Analyzer(
     private def lookupTableFromCatalog(
         u: UnresolvedRelation,
         defaultDatabase: Option[String] = None): LogicalPlan = {
+      val tableIdentWithDb = u.tableIdentifier.copy(database = u.tableIdentifier.database
```

Comment:
please follow the existing style: https://github.com/apache/spark/pull/16674/files#diff-57b3d87be744b7d79a9beacf8e5e5eb2L622